optuna/visualization.py

```diff
@@ -22,11 +22,11 @@
     import plotly.graph_objs as go
     from plotly.graph_objs._figure import Figure  # NOQA
     from plotly.subplots import make_subplots
-    _available = True
+    available = True
```

Comment:
[needs discussion]
I'm a bit negative about this naming because it gets ambiguous when public.
How about making this more verbose like `plotly_available` or `vis_available`? (I know these names are not pretty)`

Reply:
If it's the case that plotly is to be exposed through some of our public APIs (we should probably discuss this first) as our visualization backend (and it makes sense to expose its name) then your first suggestion sounds good to me. It's also consistent with the check function and they probably should be.

Reply:
OK, it makes sense.

After giving consideration, I think it'd be better to expose `available` via another API such as `optuna.visualization.is_available()` (`check_plotly_availability()` is OK) in order to explicitly prohibit `optuna.visualization.available = False`.

Reply:
I like `optuna.visualization.is_available()`.

Reply:
Thanks for the inputs, I changed the interface to the one above. I also tried to align the check function but am not sure about its behavior (it raises an `ImportError`). Actually, one option would be to keep this check function internal and as the availability check is sufficient.

Reply:
> Actually, one option would be to keep this check function internal and as the availability check is sufficient.

As of now, `check_available()` alike `is_available()` from their names but they behave differently (`is_available()` returns boolean while `check_available()` raises ImportError if not `_available`).
So, to avoid confusion, I would like `check_available` to be private.

Reply:
From my experience, "check" communicates to the caller that it either does nothing or raises an error, but I'm not sure this is a more general consensus. Similar to how "is" communicates that it's boolean.

Reply:
summary of the offline discussion.

In the shoes of those who conditionally use `optuna.visualization` APIs according to its availability...

If `is_available()` only is public, they can write the code to plot results if and only if the dependency, i.e., plotly, is installed and will not suggest other users install it.
However, if `check_available()` is also public, by calling it at the very beginning of scripts, users can strongly recommend others to install plotly.

Reply:
We in my understand decided to only keep `is_available()` as it makes the APIs simple and sufficient for doing whatever check needed. I'll update the PR but let me know if I've misunderstood.

Reply:
Yes, 100% correct.
Thank you for your cooperation, @g-votte.